fast64_internal/oot/oot_utility.py

```diff
@@ -602,3 +633,279 @@ def oot_utility_register():
 def oot_utility_unregister():
 	for cls in reversed(oot_utility_classes):
 		unregister_class(cls)
+
+def getAndFormatActorParameterPart(object, field, shift, mask):
+	'''Returns the actor parameter with the shifts'''
+	attr = getattr(object, field, '0x00')
+	if attr != '0x00' and attr is not None:					
+		if isinstance(attr, str):
+			if shift != '0':
+				return f"(({attr} << {shift}) & {mask})"
+			else:
+				return f"({attr} & {mask})" 
+		elif isinstance(attr, bool):
+			if attr:
+				return f"(1 << {shift})"
+			else:
+				return None
+		else:
+			raise NotImplementedError
+	else:
+		return None
+
+def getActorFinalParameters(detailedProp, actorKey, paramTarget, field):
+	'''Returns the full parameter string'''
+	params = []
+	paramPart = []
+	if field is not None:
+		panelParams = getattr(detailedProp, field, "")
+		if panelParams == "":
+			return "0x0"
+	for actorNode in root:
+		if actorKey == actorNode.get('Key'):
+			lenProp = getLastElemIndex(actorKey, 'Property', None)
+			lenSwitch = getLastElemIndex(actorKey, 'Flag', 'Switch')
+			lenBool = getLastElemIndex(actorKey, 'Bool', None)
+			lenEnum = getLastElemIndex(actorKey, 'Enum', None)
+			for elem in actorNode:
+				actorType = getattr(detailedProp, actorKey + '.type', None)
+				if hasTiedParams(elem.get('TiedParam'), actorType):
+					paramPart = getActorString(elem, detailedProp, actorKey, lenProp, lenSwitch, lenBool, lenEnum, paramTarget)
+					if paramPart is not None:
+						params += paramPart
+	actorProps = " | ".join(params)
+	if paramTarget == 'Params':
+		actorType = getattr(detailedProp, actorKey + '.type', '0')
+		if len(params) == 0:
+			return f'0x{actorType}'
+		else:
+			return f'(0x{actorType} | ({actorProps}))'
+	elif len(params) > 0:
+		return actorProps
+	else:
+		return '0x0'
+
+def setActorParameter(object, field, param, mask):
+	'''Sets the attributes to have the correct display on the UI'''
+	shift = len(f'{mask:016b}') - len(f'{mask:016b}'.rstrip('0'))
+	if field.endswith('.type'):
+		setattr(object, field, f'{param & mask:04X}')
+	else:
+		attr = getattr(object, field, '0x0')
+		if isinstance(attr, str):
+			setattr(object, field, f'0x{(param & mask) >> shift:02X}')
+		elif isinstance(attr, bool):
+			setattr(object, field, bool((param & mask) >> shift))
+		else:
+			raise NotImplementedError
+
+def getLastElemIndex(actorKey, elemTag, flagType):
+	'''Looking for the last index of an actor's property (from XML data)'''
+	indexes = []
+	for actorNode in root:
+		if actorNode.get('Key') == actorKey:
+			for elem in actorNode:
+				if elem.tag == elemTag:
+					if flagType is None or (flagType == 'Switch' and elem.get('Type') == flagType):
+						indexes.append(elem.get('Index'))
+	indexes.sort()
+	return f"{indexes[len(indexes) - 1]}" if len(indexes) != 0 else "0"
```

Comment:
`indexes[len(indexes) - 1]` can be just `indexes[-1]` in python, but moreover:
```suggestion
	return max(indices) if indices else 0
```
also idk about 0 being the default
also idk if the function should check all indices are >0 ?
(I didn't entirely understand what this is for yet)